File: LeetCode/Java/Easy/PlusOne.java
package Easy;

public class PlusOne {
	
	public void main(String[] args){
		
	}
	
	public int[] plusOne(int[] digits){
		int len = digits.length;
		
		for(int i=len-1; i>=0; i--){
			if(digits[i] < 9){
				digits[i]++;
				return digits;
			}
			digits[i]=0;
		}
		int[] sum = new int[len+1];
		sum[0] = 1;
		
		return sum;
	}

}

File: LeetCode/Python/Easy/PlusOne.py
class Solution:
    def plusOne(self, digits):
        length = len(digits) - 1
        while length >= 0:
            if digits[length] < 9:
                digits[length] += 1
                return digits
            elif digits[length] == 9:
                digits[length] = 0
                if length == 0:
                    digits.insert(0, 1)
            length -= 1
        return digits
